src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -222,5 +222,51 @@ internal static void AddCustomAttributesToExtensions(this IDictionary<string, IO
 
             return atrributesValueMap;
         }
+
+        /// <summary>
+        /// Checks whether the base type of an <see cref="IEdmStructuredType"/> is referenced as a type within the Edm model.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="baseType">The base type of the target <see cref="IEdmStructuredType"/>.</param>
+        /// <param name="structuredTypes">Optional: The IEnumerable of <see cref="IEdmStructuredType"/> to check against.</param>
+        /// <param name="actions">Optional: The IEnumerable of <see cref="IEdmAction"/> to check against.</param>
+        /// <returns>True if reference is found, otherwise False.</returns>
+        internal static bool IsBaseTypeReferencedAsTypeInModel(
+            this IEdmModel model,
+            IEdmStructuredType baseType,
+            IEnumerable<IEdmStructuredType> structuredTypes = null,
+            IEnumerable<IEdmAction> actions = null)
+        {
+            const string Entity = "entity";
+            string baseTypeName = baseType?.FullTypeName();
+            bool isBaseTypeEntity = Entity.Equals(baseTypeName?.Split('.').Last(), StringComparison.OrdinalIgnoreCase);
+
+            if (!string.IsNullOrEmpty(baseTypeName) && !isBaseTypeEntity)
+            {
+                structuredTypes ??= model.GetAllElements()
+                        .Where(x => x.SchemaElementKind == EdmSchemaElementKind.TypeDefinition)
+                        .Where(y => !y.Name.Equals(Entity, StringComparison.OrdinalIgnoreCase))
+                        .OfType<IEdmStructuredType>();
+
+                actions ??= model.GetAllElements()
+                        .Where(x => x.SchemaElementKind == EdmSchemaElementKind.Action)
+                        .OfType<IEdmAction>();
+
+                // Is base type referenced as a type in any property within a structured type
+                bool isReferencedInStructuredType = structuredTypes
+                    .Any(x => x.DeclaredProperties.Where(x => x.Type.TypeKind() == EdmTypeKind.Entity)
```

Comment:
don't you get an error saying variable x is already in use?

Reply:
Yes, good catch! It should be:
```
if (context.Settings.EnableDefaultValueForOdataTypeProperty ||
    (!context.Settings.EnableDefaultValueForOdataTypeProperty &&
    (isBaseTypeAbstractNonEntity ||   
  context.Model.IsBaseTypeReferencedAsTypeInModel(structuredType.BaseType))))
```

Reply:
Nope

Reply:
I believe what @baywet is saying here is since the anonymous function is not static, it would be better to create a distinction between the `x` in the nested function and the outer one.